### ISSUE TITLE
Inconsistency in file naming under headers directory

### ISSUE THREAD

**Issue by Kevin-Jin** [Rblp/blp#3]
I'm currently building a configure.win file for Rblpapi. The header files are downloaded from
`https://github.com/Rblp/blp/raw/master/headers/${platform}/blpHeaders.tar.gz`
While the shared library is downloaded from
`https://github.com/Rblp/blp/raw/master/${platform}${flavour}/blpLibrary.tar.gz`

Either /win32 and /win64 need to be renamed to /windows32 and /windows64 or /headers/windows needs to be renamed to /headers/win.

**Comment by Kevin-Jin** [comment]
Just realized Makevars.win handles this so I didn't need to create a configure.win.

**Comment by eddelbuettel** [comment]
What makes you say that, or exactly what are you trying to say?

This has worked ever since we put the package onto CRAN.  And it does of course work unattended, via win-builder (which is how CRAN builds binaries for Windows) and now r-hub.

**Comment by Kevin-Jin** [comment]
My mistake. This is an non-issue because in Makevars.win, line 46 downloads:
`https://github.com/Rblp/blp/raw/master/headers/windows/blpHeaders.tar.gz`
While line 47 downloads:
`https://github.com/Rblp/blp/raw/master/win${WIN}/blpLibrary.tar.gz`
Makevars just behaves differently on non-Windows machines.

**Comment by eddelbuettel** [comment]
Well, yes, that is how R works:  `configure` and `src/Makevars` for real computers.  `configure.win` and `src/Makevars.win` for Windoze.  See "Writing R Extensions".  

So you're able to build locally now?  Good!

**Comment by Kevin-Jin** [comment]
After fixing my PATH to include Rtools and using joel23888's fixes for blpapi_call.h in Rblpapi's issue 164, yep!

```
53c53
<     (throw UnsupportedOperationException(#FUNCNAME " not supported"), 0)
---
>     (throw UnsupportedOperationException(#FUNCNAME " not supported"))
77c77
<       : 0)
---
>       : (void) 0)
280c280
<       : 0)
---
>       : (void) 0)
```

**Comment by eddelbuettel** [comment]
Could you just try to be a bit more specific and and helpful, please? Referring to a thread predating subsequent releases _and_ changes to the `blp` repo here doesn't help.

I don't mean to berate you here but have failed to
- show what "broke"
- show what you changed to "fix" this

when at the same we have builds on CRAN demonstrating that, well, nothing appears to be broken.

**Comment by Kevin-Jin** [comment]
Sorry that my responses have been curt and unintelligible. I just didn't want to waste your time with my own mistakes. The original post should be ignored because it was a simple issue with my build environment that I trivially fixed.

As for the issue I raised in my previous response, I honestly don't know why the build failed for me. Perhaps the compiler on CRAN is less strict about type safety because I simply needed to add casts to resolve the issues. Perhaps it may help to contact joel23888 directly. Again, his posts that were helpful to me are at https://github.com/Rblp/Rblpapi/issues/164#issuecomment-211712399 and https://github.com/Rblp/Rblpapi/issues/164#issuecomment-211635475.

For the sake of expediency in resolving the original enhancement request, https://github.com/Rblp/Rblpapi/issues/215, I won't be pursuing the causes of the header file issues at this moment but I may return to them. To be more specific, I basically just opened the parent directory of where I cloned Rblpapi and then ran:
`R CMD INSTALL Rblpapi`
I let the build fail after it extracted blpapi_call.h to inst/include. Then I opened inst/include/blpapi_call.h, removed the 0 in the second argument to throw UnsupportedOperationException in the BLPAPI_UNSUPPORTED_CALL macro, casted the 0 to void in the BLPAPI_CALL_SESSIONOPTIONS_SETMAXEVENTQUEUESIZE macro, and casted the 0 to void in the BLPAPI_CALL_SESSIONOPTION_SETRECORDSUBSCRIPTIONDATARECEIVETIMES macro. Then I simply reran:
`R CMD INSTALL Rblpapi`
And everything works fine now. I'm using g++ version 4.6.3. In order to build on Windows, I use Rtools version 3.3.0.1958.

Hope this helps.
